dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/DataSourceServiceImpl.java

```diff
@@ -645,6 +643,55 @@ public Map<String, Object> getTableColumns(Integer datasourceId, String tableNam
         return result;
     }
 
+    @Override
+    public Map<String, Object> getDatabases(Integer datasourceId) {
```

Comment:
could we directly return `List<ParamsOptions>` and wrapper to `Result` in the controller?

Reply:
No, If occurs some issue, will return like `putMsg(result, Status.GET_DATASOURCE_TABLES_ERROR);`

Reply:
could we directly throw exception instead of wrapper into hashmap?

Reply:
Yes, we can. But looks like not a simple job, because you can see we use map in all data source services. We can update it in the future